### PR TITLE
Shogun Python package using setup.py

### PR DESCRIPTION
**Pull request by arianepaola**
Shogun Python package using setup.py

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    def run(self):
+        print('RUN INSTALL')
+        
+        shogun_cmake() 
+        chdir(getcwd())
+        self.do_egg_install() 
+
+shogun_setup(
+    name = 'shogun',
+    version = '4.2.0',
```
Comment (vigsterkr) [inline setup.py]: this should be generated somehow....

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    keywords = 'Large scale kernel methods and Support Vector Machines (SVM)',
+    
+    zip_safe = False,
+    
+    cmdclass = {'build': ShogunBuild, 'install': ShogunInstall},
+    
+    packages = ['shogun'],
+
+    install_requires = ['numpy'],
```
Comment (vigsterkr) [inline setup.py]: i reckon we require much more python packages than numpy.... :S

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    
+    def run(self):
+        shogun_cmake(self.cmake)
+    
+    # Command.sub_commands
+
+
+class ShogunInstall(setuptools.command.install.install):
+    def run(self):
+        print('RUN INSTALL')
```
Comment (lisitsyn) [inline setup.py]: Can we get more meaningful message? ;)

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    plat_name = ''
+
+    def initialize_options(self):
+        self.cmake = None
+    
+    def finalize_options(self):
+        pass
+    
+    def run(self):
+        shogun_cmake(self.cmake)
```
Comment (lisitsyn) [inline setup.py]: Why no message for build but some message for install?

**Comment by vigsterkr** [comment]
@arianepaola let's get this moving...

**Comment by arianepaola** [comment]
@vigsterkr @lisitsyn @karlnapf @lgoetz I have mentioned on IRC that I did strip down this version to a minimum to create a pull request.

File: setup.py (hunk 1 above)
Comment (arianepaola) [inline setup.py]: I am trying to get the information from CMake when bootstrapping the version generation to get the string.

File: setup.py (hunk 2 above)
Comment (arianepaola) [inline setup.py]: I know :smile: it is very minimal at the moment, but the dependencies will be added.

File: setup.py (hunk 3 above)
Comment (arianepaola) [inline setup.py]: sure :+1:

File: setup.py (hunk 4 above)
Comment (arianepaola) [inline setup.py]: Work in progress :smile:

Comment (arianepaola) [inline setup.py]: @lisitsyn I have added them in a new version.

File: setup.py (hunk 1 above)
Comment (arianepaola) [inline setup.py]: @vigsterkr Done in the new version.

File: setup.py (hunk 2 above)
Comment (arianepaola) [inline setup.py]: @vigsterkr Let's update the dependencies. I have published a final version.

**Comment by arianepaola** [comment]
@vigsterkr @lisitsyn @karlnapf @lgoetz  Please review, comment and test. It is fully functional and requires only minor tweaks before merging.

**Comment by arianepaola** [comment]
@vigsterkr @lisitsyn @karlnapf @lgoetz Please review, comment and test. It is fully functional and requires only minor tweaks before merging.

**Comment by vigsterkr** [comment]
ok so couple of things:
 ... does this require it to run cmake prior to run setup.py, because if so it's a problem... and why do we think that that's required

when i checkout a clean repo and do `python setup.py --version`
i'm getting this:
```
shogun wiking$ python setup.py --version
Retrieving Shogun version
Verifying Shogun preconditions
Shogun build environment completed tasks: cmake: [False] -  make: [False] - make install: [False]
Bootstrapping Shogun
Running CMake
CMake arguments: -DPythonModular=ON -DENABLE_TESTING=OFF -DCMAKE_INSTALL_PREFIX=install
Creating build directory: /Users/wiking/backup/shogun/build
-- The C compiler identification is AppleClang 7.0.2.7000181
-- The CXX compiler identification is AppleClang 7.0.2.7000181
CMake Error at /usr/local/Cellar/cmake/3.5.2/share/cmake/Modules/Platform/Darwin.cmake:76 (message):
  CMAKE_OSX_DEPLOYMENT_TARGET is '10.10' but CMAKE_OSX_SYSROOT:

   ""

  is not set to a MacOSX SDK with a recognized version.  Either set
  CMAKE_OSX_SYSROOT to a valid SDK or set CMAKE_OSX_DEPLOYMENT_TARGET to
  empty.
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.5.2/share/cmake/Modules/CMakeSystemSpecificInformation.cmake:36 (include)
  CMakeLists.txt:27 (project)


-- Configuring incomplete, errors occurred!
```

**Comment by arianepaola** [comment]
@vigsterkr As discussed on IRC:
It is not needed to manually run cmake prior to running setup.py.
setup.py will run a bootstrapping process of  cmake, make and make install, as it requires the generated files / content for the Python package.

As Viktor mentioned, cmake was installed using brew. The setup.py script uses the default system cmake and might not have picked up the updated user's PATH.

**Comment by arianepaola** [comment]
@vigsterkr I squashed the commits.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+        'Intended Audience :: Education',
+        'Intended Audience :: Information Technology',
+        'Intended Audience :: Science/Research',
+        'Topic :: Software Development :: Libraries',
+        # Python 2 and Python 3 support
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 3'
+    ],
+    
+    keywords = 'Large scale kernel methods and Support Vector Machines (SVM)',
```
Comment (karlnapf) [inline setup.py]: Can you put our updated description?
"Unified and efficient Machine Learning"

https://github.com/shogun-toolbox/shogun/wiki/Project-description

**Comment by arianepaola** [comment]
@karlnapf Updated description.
@vigsterkr ready to merge

**Comment by karlnapf** [comment]
Fine to merge.

**Pull request merged**